UnitTests/MPKitContainerTests.m

```diff
@@ -2554,6 +2581,507 @@ - (void)testAppInfoContainsSideloadKitsFlag {
     XCTAssertEqualObjects(dict[@"sideloaded_kits_count"], @3);
 }
 
+- (void)testFilterEventTypeForSideloadedKit {
+    MPKitTestClassSideloaded *sideloadedKitInstance1 = [[MPKitTestClassSideloaded alloc] init];
+    MPSideloadedKit *sideloadedKit1 = [[MPSideloadedKit alloc] initWithKitInstance:sideloadedKitInstance1];
+    [sideloadedKit1 addEventTypeFilterWithEventType:MPEventTypeOther];
+    kitContainer.sideloadedKits = @[sideloadedKit1];
+    [kitContainer initializeKits];
+    
+    MPEvent *event = [[MPEvent alloc] initWithName:@"Dinosaur Run" type:MPEventTypeOther];
+    event.duration = @2;
+    event.customAttributes = @{@"speed":@25,
+                   @"modality":@"sprinting"};
+    event.category = @"Olympic Games";
+
+    NSSet<id<MPExtensionProtocol>> *registeredKits = [MPKitContainer registeredKits];
+    id registeredKit = [[registeredKits objectsPassingTest:^BOOL(id<MPExtensionProtocol>  _Nonnull obj, BOOL * _Nonnull stop) {
+        if ([obj conformsToProtocol:@protocol(MPExtensionKitProtocol)]) {
+            id<MPExtensionKitProtocol> kitExtension = (id<MPExtensionKitProtocol>)obj;
+            if (kitExtension.code.intValue == 1000000000) {
+                return YES;
+            }
+        }
+        return NO;
+    }] anyObject];
```

Comment:
```suggestion
    NSSet<id<MPExtensionProtocol>> *registeredKits = [MPKitContainer registeredKits];
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"code == 1000000000"];
    id registeredKit = [[registeredKits filteredSetUsingPredicate:predicate] anyObject];
```

Seems like this does the same thing, but is much easier to read and less lines of code. I see it's used in the other tests further down so would be nice to standardize on it.

Reply:
sorry about that I thought of a better way halfway through and didn't update the older tests